File: app/src/main/java/miui/statusbar/lyric/QuickTitleService.java
package miui.statusbar.lyric;

import android.annotation.TargetApi;
import android.graphics.drawable.Icon;
import android.os.Build;
import android.service.quicksettings.Tile;
import android.service.quicksettings.TileService;
import android.util.Log;
import miui.statusbar.lyric.config.Config;
import miui.statusbar.lyric.utils.ActivityUtils;
import miui.statusbar.lyric.utils.Utils;

@TargetApi(Build.VERSION_CODES.N)
public class QuickTitleService extends TileService {
    static Tile tile;

    //    @Override
//    public void onCreate() {
//        super.onCreate();
//        Context context = getBaseContext();
//        Tile tile = getQsTile();
//        Config config = ActivityUtils.getConfig(context);
//        set(tile, config);
//    }
//
    @Override
    public void onClick() {
        super.onClick();
        Config config = ActivityUtils.getConfig(getBaseContext());
        assert config != null;
        config.setLyricService(!config.getLyricService());
        set(config);
    }

    private void set(Config config) {
        tile.setIcon(Icon.createWithResource(this, R.drawable.title_icon));
        tile.setLabel(getString(R.string.QuickTitle));
        tile.setContentDescription(getString(R.string.QuickTitle));
        tile.setState(config.getLyricService() ? Tile.STATE_ACTIVE : Tile.STATE_INACTIVE);
        tile.updateTile();
    }

    @Override
    public void onStartListening() {
        super.onStartListening();
        Config config = ActivityUtils.getConfig(getBaseContext());
        tile = getQsTile();
        assert config != null;
        Log.i("", String.valueOf(config.getLyricService()));
        Utils.showToastOnLooper(getBaseContext(), config.getLyricService() + "");
        set(config);
    }
}

File: app/src/main/java/cn/fkj233/xposed/statusbarlyric/service/QuickTitleService.kt
package cn.fkj233.xposed.statusbarlyric.service

import android.graphics.drawable.Icon
import android.service.quicksettings.Tile
import android.service.quicksettings.TileService
import cn.fkj233.xposed.statusbarlyric.R
import cn.fkj233.xposed.statusbarlyric.config.Config
import cn.fkj233.xposed.statusbarlyric.utils.Utils

class QuickTitleService: TileService() {
    private lateinit var tile: Tile

    override fun onClick() {
        super.onClick()
        val config: Config? = Utils.getSP(baseContext, "Lyric_Config")?.let { Config(it) }
        config?.setLyricService(!config.getLyricService())
        config?.let { set(it) }
    }

    fun set(config: Config) {
        tile.icon = Icon.createWithResource(this, R.drawable.title_icon)
        tile.label = getString(R.string.QuickTitle)
        tile.contentDescription = getString(R.string.QuickTitle)
        tile.state = if (config.getLyricService()) Tile.STATE_ACTIVE else Tile.STATE_INACTIVE
        tile.updateTile()
    }

    override fun onStartListening() {
        super.onStartListening()
        val config: Config? = Utils.getSP(baseContext, "Lyric_Config")?.let { Config(it) }
        tile = qsTile
        config?.let { set(it) }
    }
}

